src/Compilers/CSharp/Portable/Lowering/ExtensionMethodReferenceRewriter.cs

```diff
@@ -194,8 +194,7 @@ method.OriginalDefinition is ErrorMethodSymbol ||
 
         public override BoundNode? VisitMethodDefIndex(BoundMethodDefIndex node)
         {
-            MethodSymbol method = node.Method;
-            Debug.Assert(method.IsDefinition); // Tracked by https://github.com/dotnet/roslyn/issues/78962 : From the code coverage and other instrumentations perspective, should we remap the index to the implementation symbol? 
+            MethodSymbol method = VisitMethodSymbolWithExtensionRewrite(this, node.Method);
```

Comment:
>VisitMethodSymbolWithExtensionRewrite [](http://example.com/codeflow?start=34&length=37)

Instead of changing accessibility of this method, I think we should extract the body of the enclosing method into a helper that body rewriter can call. See how method below does that.